[annabellekk] iP

annabellekk opened the pull request:
## Dawn
> “The secret of getting ahead is getting started.” – Mark Twain

Stop procrastinating, get started with your to-dos, and Dawn is here to help you with that! It's
- text-based
- easy to learn
- ~~FAST SUPER~~ FAST to use

All you need to do is,
1. download it from [here](https://github.com/annabellekk/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 
And it is **FREE**! 🤩 

Features:
- [X] Managing tasks
- [X] Searching tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```Java
public static void main(String[] args) {
    try {
        new Dawn("data/Dawn.txt").run();
    } catch (DawnException e) {
        System.out.println(e);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Renamed main file and added new greetings

Commit:
Add support for echo

Commit:
Added add and listing functions

Commit:
Added marking functionality

Commit:
Added Task class

Commit:
Added different task types

Commit:
added automated I/O testings

Commit:
Handle empty description and incorrect input errors

Commit:
Handle errors for missing deadlines and start and end times

Commit:
Added delete tasks function and corresponding error handling

Commit:
Added new functions to handle long code, and used enum

Commit:
Fixed bugs in the functions and handled edge cases

Commit:
Correct indentations of switch statements

Commit:
Add method to load previously saved tasks

Commit:
Add method to save tasks

Commit:
Handle missing file and directory cases

Commit:
Merge branch 'branch-Level-7'Add functions to load previously saved tasks and save current task lists

Commit:
Store deadline dates as LocalDate format

Commit:
Add function to print tasks occuring today

Commit:
Merge branch-Level-8: Completed Level 8 increment

Commit:
Split functions to separate classes adhering to OOP

Commit:
Handle exceptions thrown

Commit:
Merge branch 'branch-A-MoreOOP': Extracted classes out to adhere to OOP principles

Commit:
Add packaging

Commit:
Merge branch 'add-gradle-support': A-Gradle

Commit:
Add JUnit test cases for Parser class

Commit:
Add test cases for TaskList class

Commit:
Merge branch 'branch-JUnit': Added test cases to test Parser class and TaskList class

Commit:
Alter file names in build.gradle

Commit:
Add JavaDoc comments for all public methods

Commit:
Adjust code to follow coding standard

Commit:
Add find tasks method

Commit:
Merge branch 'branch-A-JavaDoc': Added JavaDoc comments to all public methods

Commit:
Resolve merge conflicts

Commit:
Resolve merge conflicts

Commit:
Fix bugs

tanyqcollin commented on src/main/java/Dawn/DawnException.java:
```diff
@@ -0,0 +1,20 @@
+package Dawn;
+
+public class DawnException extends Exception{
+    String msg;
```
should use `message` to make code more understandable.

annabellekk replied:
Noted!

tanyqcollin commented on src/main/java/Dawn/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    String deadline;
+    LocalDate date;
```
should declare access modifier to make it clear is this variable are intended to be accessible by other classes or packages and what parts are meant to be internal. You may declare it as `private` attribute as you have a getter for the deadline.

annabellekk replied:
Thank you for the comment, I missed this out

tanyqcollin commented on src/main/java/Dawn/Parser.java:
```diff
@@ -0,0 +1,113 @@
+public class Parser {
+    private String command;
+    private String input;
+    private enum Command {
+        TODO,
+        DEADLINE,
+        EVENT,
+        MARK,
+        UNMARK,
+        DELETE,
+        BYE,
+        LIST,
+        TODAY,
+        FIND
+    }
```
nice use of the enum

tanyqcollin commented on src/main/java/Dawn/ToDo.java:
```diff
@@ -0,0 +1,16 @@
+package Dawn;
+public class ToDo extends Task {
```
appropriate use of inheritance

tanyqcollin reviewed (commented):
LGTM. The code looks good for to me and Javadoc comments are presented as well, no major issues.

jonushzw commented on src/main/java/Dawn/Dawn.java:
```diff
@@ -0,0 +1,39 @@
+package Dawn;
+public class Dawn {
```
Could add javadocs to the class as well!

annabellekk replied:
Ok! Thanks for the suggestion, I'll add them in

jonushzw commented on src/main/java/Dawn/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+
+public class Deadline extends Task {
+    String deadline;
+    LocalDate date;
```
should add access modifier for this

jonushzw commented on src/main/java/Dawn/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    @Override
+    public String toString() {
```
Good job formatting the toString!

jonushzw commented on src/main/java/Dawn/Parser.java:
```diff
@@ -0,0 +1,113 @@
+            System.out.println(e);
+        }
+    }
+
```
Great use of switch statements!

annabellekk replied:
Thank you :)

jonushzw commented on src/main/java/Dawn/Ui.java:
```diff
@@ -0,0 +1,52 @@
+     * @throws DawnException
+     */
+    public void greet() throws DawnException {
+        String divider = "--".repeat(30);
```
Great use of repeat method! Did not think of that for my own code 👍

jonushzw reviewed (commented):
Great job!

Commit:
Add checkstyle tools in gradle.build

Commit:
Add javaFx files

Commit:
Set up javafx

Commit:
Complete JavaFX tutorial parts1-3

Commit:
Complete JavaFX part4

Commit:
Update ui

Commit:
Add GUI to chatbot

Commit:
Merge branch 'branch-Level-10': Added GUI for the chatbot

Commit:
Change add tasks methods

There were two add tasks methods to handle the difference in format of
the tasks that were stored from the previous use vs the tasks entered
into the command line by the user.

Since these two methods have the same logic, let's merge them together
so we don't have duplicate code.

Modifying the format in which tasks are stored so that they can be parsed
easily, and modifying the way in which tasks are parsed to allow alternate
formats.

Commit:
Use assertions

Using java assertions to define assumptions such that potential
bugs can be spotted easily.

annabellekk committed:
Merge pull request #3 from annabellekk/branch-A-Assertions

Use assertions

Commit:
Merge branch 'master' into branch-A-CodeQuality

Encountered conflicts when merging master branch to
branch-A-CodeQuality as was working on two different
branches concurrently.

annabellekk committed:
Merge pull request #2 from annabellekk/branch-A-CodeQuality

Change add tasks methods

Commit:
Add reminder method

Commit:
Merge branch 'branch-B-Reminders'

Add reminder method to allow users to receive reminder for all upcoming
deadlines and events.

Commit:
Change package name to adhere to coding standard

Commit:
Remove dead code in parserTest file

Commit:
Change gradle to reflect updated package names

Commit:
Fix bug in delete command

Commit:
Change profile pictures of user and Dawn

annabellekk committed:
Add a screenshot of Dawn in action

Commit:
Test the output of Dawn

Commit:
Merge branch 'master' of https://github.com/annabellekk/ip

annabellekk committed:
Update README.md

annabellekk committed:
Update README.md

annabellekk committed:
Update README.md

Commit:
Fix bug in findTask method

Commit:
Merge branch 'master' of https://github.com/annabellekk/ip

Commit:
Fix bug with remind method

annabellekk committed:
Update README.md

annabellekk committed:
Update README.md

annabellekk committed:
Update README.md

annabellekk committed:
Update README.md

annabellekk committed:
Update README.md

Commit:
Fix bugs in handling invalid input dates

Commit:
Fix invalid date error message

annabellekk committed:
Update README.md
